### PR TITLE
TS export components with static props - fix TS2532

### PR DESCRIPTION
**Pull request by SergeyRoyt**
https://monday.monday.com/boards/3532714909/pulses/4488701805

image

### CODE REVIEW

File: src/types/withStaticProps.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import VibeComponent from "./VibeComponent";
+
+export const withStaticProps = <T, P, S>(forwarded: VibeComponent<T, P>, staticProps: S) =>
```
Comment (shlomitc) [inline src/types/withStaticProps.ts:3]: I don't really like that we need to wrap it with another higher-order function. Are you sure there isn't a better way?

Comment (SergeyRoyt) [reply src/types/withStaticProps.ts]: Tried a lot of different stuff, but haven't found anything else that's working

File: src/components/AlertBanner/AlertBanner.tsx [hunk 2]
```diff
@@ -27,7 +28,7 @@ interface AlertBannerProps extends VibeComponentProps {
   children?: ReactElement<AlertBannerButtonProps | AlertBannerLinkProps | AlertBannerTextProps>;
 }
 
-const AlertBanner: React.FC<AlertBannerProps> & {
+const AlertBanner: VibeComponent<AlertBannerProps> & {
```
Comment (shlomitc) [inline src/components/AlertBanner/AlertBanner.tsx:31]: just making sure this is intentionally

File: src/components/Avatar/Avatar.tsx [hunk 3]
```diff
@@ -202,11 +202,9 @@ const Avatar: React.FC<AvatarProps> & {
   );
 };
 
-Object.assign(Avatar, {
+export default withStaticProps(Avatar, {
```
Comment (shlomitc) [inline src/components/Avatar/Avatar.tsx:205]: this hides some issues:
1. the static properties are still optional.
2. the colors values aren't autocompleted

**Review by shlomitc — CHANGES_REQUESTED** [review]
1. consider not using `withStaticProps` if not needed (in FC)
2. there's probably an issue with how it works if it's not defined

File: src/components/AlertBanner/AlertBanner.tsx (hunk 2 above)
Comment (SergeyRoyt) [reply src/components/AlertBanner/AlertBanner.tsx:31]: It was, it will correctly pass the ref now

File: src/components/Avatar/Avatar.tsx (hunk 3 above)
Comment (SergeyRoyt) [reply src/components/Avatar/Avatar.tsx:203]: Fixed colors values

File: src/components/Button/Button.tsx [hunk 4]
```diff
@@ -383,6 +383,7 @@ const Button: VibeComponent<ButtonProps, unknown> & {
   }
 );
 
+// Can be removed when defaultProps are removed
```
[the remark this answers is not in the document]
Comment (SergeyRoyt) [reply src/components/Button/Button.tsx:386]: It's being used in Button.defaultProps

File: src/utils/colors-vars-map.ts [hunk 5]
```diff
@@ -174,194 +174,207 @@ export enum ColorStyle {
...
-  "--color-peach",
-  "--color-berry",
-  "--color-winter",
-  "--color-river",
-  "--color-navy",
-  "--color-aquamarine",
-  "--color-indigo",
-  "--color-dark_indigo"
-];
+// export const allMondayColors = [
```
[the remark this answers is not in the document]
Comment (SergeyRoyt) [reply src/utils/colors-vars-map.ts:266]: Not sure why it was needed, so didn't want to delete...

File: src/components/Button/Button.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (SergeyRoyt) [reply src/components/Button/Button.tsx:386]: We'll remove it when we'll remove Button.defaultProps

[the remark this answers is not in the document]
Comment (shlomitc) [reply src/components/Button/Button.tsx:386]: I tried to comment on lines 387-393
```tsx
Object.assign(Button, {
  sizes: SIZES,
  colors: ButtonColor,
  kinds: ButtonType,
  types: ButtonInputType,
  inputTags: ButtonInputType
});
```

[the remark this answers is not in the document]
Comment (shlomitc) [reply src/components/Button/Button.tsx:386]: It you're talking about using the static values in the default props - then simply replace them with the direct values

[the remark this answers is not in the document]
Comment (shlomitc) [reply src/components/Button/Button.tsx:386]: You mean it will show a string value instead of the enum? How does it work in other components?

**Review by shlomitc — APPROVED** [review]

**Pull request merged**